src/transaction/sign.js

```diff
@@ -6,23 +6,38 @@ const binary = require('ripple-binary-codec');
 const {computeBinaryTransactionHash} = require('ripple-hashes');
 const validate = utils.common.validate;
 
-function computeSignature(txJSON, privateKey) {
-  const signingData = binary.encodeForSigning(txJSON);
+function computeSignature(tx: Object, privateKey: string, signFor: ?string) {
+  const signingData = signFor ?
+    binary.encodeForMultisigning(tx, signFor) : binary.encodeForSigning(tx);
   return keypairs.sign(signingData, privateKey);
 }
 
-function sign(txJSON: string, secret: string
+function sign(txJSON: string, secret: string, options: Object = {}
 ): {signedTransaction: string; id: string} {
   validate.sign({txJSON, secret});
   // we can't validate that the secret matches the account because
   // the secret could correspond to the regular key
 
   const tx = JSON.parse(txJSON);
+  if (tx.TxnSignature || tx.Signers) {
+    throw new utils.common.errors.ValidationError(
+      'txJSON must not contain "TxnSignature" or "Signers" properties');
+  }
+
   const keypair = keypairs.deriveKeypair(secret);
-  if (tx.SigningPubKey === undefined) {
-    tx.SigningPubKey = keypair.publicKey;
+  tx.SigningPubKey = options.signFor ? '' : keypair.publicKey;
```

Comment:
@sublimator  "In rippled, that's definitely what it means in the 'sign_for' command" - the documentation and test results contradict this statement: https://github.com/ripple/ripple-dev-portal/compare/gh-pages...mDuo13:simplesigning#diff-546a0b8282c4e506a611d215a8bf65ddR354

Also, what does the "proxy ?" comment mean?